core/vm/instructions_test.go

```
@@ -927,3 +928,102 @@ func TestOpMCopy(t *testing.T) {
 		}
 	}
 }
+
+// Define a custom tracer to check if the ISSTATIC opcode was called
+type isStaticTracer struct {
+	called              bool
+	returnedValue       *uint256.Int
+	opcodeAfterIsStatic byte
+}
+
+func (t *isStaticTracer) OnOpcode(pc uint64, op byte, gas, cost uint64, scope tracing.OpContext, rData []byte, depth int, err error) {
+	if op == byte(ISSTATIC) {
+		t.called = true
+	}
+	if t.called && op == t.opcodeAfterIsStatic && t.returnedValue == nil {
+		st := scope.StackData()
+		t.returnedValue = (*uint256.Int).Clone(&st[len(st)-1])
```

Just thinking out loud, but could we ever end up in a scenario where the stack is actually empty even after the `ISSTATIC` call was done? Should we have an explicit check here or let it fail during runtime? 
I might be completely wrong, in case this can never happen :)

---

The latter. After `ISSTATIC`, the stack should have a new value pushed -- either `1` or `0`.  

If that’s not the case, it indicates a bug or an unexpected issue. Hence the user should be informed with a failure (panic), rather than allowing execution to continue in an undefined state.